imports/ui/dialogs/unit-select-dialog.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { createContainer } from 'meteor/react-meteor-data'
+import PropTypes from 'prop-types'
+import Dialog from 'material-ui/Dialog'
+import { push } from 'react-router-redux'
+import FontIcon from 'material-ui/FontIcon'
+import { modalTitleStyle, closeDialogButtonStyle } from './generic-dialog.mui-styles'
+import UnitTypeIcon from '../unit-explorer/unit-type-icon'
+import MenuItem from 'material-ui/MenuItem'
+import { resetMenuItemDivStyle } from '../general.mui-styles'
+
+import { Meteor } from 'meteor/meteor'
+import Units, { collectionName } from '../../api/units'
+import TextField from 'material-ui/TextField'
+
+class UnitSelectDialog extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      searchMode: false,
+      searchText: '',
+      searchResult: []
+    }
+  }
+
+  onSearchChanged = (searchText) => {
+    this.setState({searchText})
+    if (searchText === '') {
+      this.setState({searchMode: false})
+    } else {
+      this.setState({searchMode: true})
+      const matcher = new RegExp(searchText, 'i')
+      const searchResult = this.props.unitList
+        .filter(unit => !matcher || (unit.name && unit.name.match(matcher)))
+      this.setState({
+        searchResult: searchResult
+      })
+    }
+  }
+
+  render () {
+    const { show, onDismissed, inProgress, dispatch, unitList, item } = this.props
+    const { searchText, searchMode, searchResult } = this.state
+    const units = searchMode ? searchResult : unitList
+    const title =
+      <div>
+        <div className='pb1'>Select Unit</div>
+        <span className='ba b--moon-gray mv1 br1 pv1'>
```

Comment:
This should be taken out of the title. The title should be just the title, and the rest of the elements should be a part of the body.